File: 0001-0500/0012/0012.integer-to-roman.cpp
/*
 * @lc app=leetcode id=12 lang=cpp
 *
 * [12] Integer to Roman
 */

// @lc code=start
class Solution
{
public:
    string intToRoman(int num)
    {
        string res;
        vector<int> nums = {1000, 900, 500, 400, 100,
                            90, 50, 40, 10, 9, 5, 4, 1};
        vector<string> romans = {"M", "CM", "D", "CD", "C",
                                 "XC", "L", "XL", "X", "IX", "V", "IV", "I"};
        for (int i = 0; i < nums.size(); i++)
        {
            while (num >= nums[i])
            {
                num -= nums[i];
                res += romans[i];
            }
        }
        return res;
    }
};
// @lc code=end

File: 0001-0500/0012/0012.integer-to-roman.py
#
# @lc app=leetcode id=12 lang=python3
#
# [12] Integer to Roman
#

# @lc code=start
val = [1000,900,500,400,100,90,50,40,10,9,5,4,1]
rom = ["M","CM","D","CD","C","XC","L","XL","X","IX","V","IV","I"]
class Solution:
    def intToRoman(self, num: int) -> str:
        res = ''
        for i in range(len(val)):
            while num >= val[i]:
                num -= val[i]
                res += rom[i]
        return res
# @lc code=end

